api/lql.go

```diff
@@ -172,75 +65,78 @@ func (q LQLQuery) ValidateRange(allowEmptyTimes bool) (err error) {
 	return nil
 }
 
-type LQLQueryResponse struct {
-	Data    []LQLQuery `json:"data"`
-	Ok      bool       `json:"ok"`
-	Message string     `json:"message"`
-}
-
-// LQLService is a service that interacts with the LQL
+// QueryService is a service that interacts with the Queries
 // endpoints from the Lacework Server
-type LQLService struct {
+type QueryService struct {
 	client *Client
 }
 
-func (svc *LQLService) Create(query string) (
-	response LQLQueryResponse,
+func (svc *QueryService) Create(nq NewQuery) (
+	response QueryResponse,
 	err error,
 ) {
-	lqlQuery := LQLQuery{QueryBlob: query}
-	if err = lqlQuery.Validate(true); err != nil {
-		return
-	}
-
-	err = svc.client.RequestEncoderDecoder("POST", apiLQL, lqlQuery, &response)
+	err = svc.client.RequestEncoderDecoder("POST", apiV2Queries, nq, &response)
 	return
 }
 
-func (svc *LQLService) Update(query string) (
-	response LQLQueryResponse,
+func (svc *QueryService) Update(queryID string, uq UpdateQuery) (
+	response QueryResponse,
 	err error,
 ) {
-	lqlQuery := LQLQuery{QueryBlob: query}
-	if err = lqlQuery.Validate(true); err != nil {
+	if queryID == "" {
+		err = errors.New("query ID must be provided")
 		return
 	}
-
-	err = svc.client.RequestEncoderDecoder("PATCH", apiLQL, lqlQuery, &response)
+	err = svc.client.RequestEncoderDecoder(
+		"PATCH",
+		fmt.Sprintf("%s/%s", apiV2Queries, url.QueryEscape(queryID)),
+		uq,
+		&response,
+	)
 	return
 }
 
-func (svc *LQLService) GetQueries() (LQLQueryResponse, error) {
-	return svc.GetByID("")
+func (svc *QueryService) GetQueries() (
```

Comment:
We adopted the function name`List()` for listing all "resources" rather than `GetResources()`.

Example: `client.V2.AgentAccessTokens.List()` vs `client.V2.AgentAccessTokens.GetAgentAccessTokens()`

Reply:
Did we adopt `Show()` for listing single "resource" ?

Reply:
Answer:
`GetByID()`